### PR TITLE
Update typing extension

### PR DESCRIPTION
**Pull request by naveen521kk**

### CODE REVIEW

**Comment by naveen521kk** [comment]
I followed [archlinux here](https://github.com/archlinux/svntogit-community/blob/packages/python-typing_extensions/trunk/PKGBUILD) but the tool `python-install` doesn't support `--prefix`/`--root` arguments so changed to `pip` as a temporary workaround. See https://github.com/FFY00/python-install/issues/8

Also, `pyproject2setuppy` doesn't work as expected for this package (not sure why), that's why using `build` + `pip` to build the package here.

**Comment by lazka** [comment]
I don't think wheels support any kind of prefix/root. They just get extracted as is.

**Comment by naveen521kk** [comment]
> I don't think wheels support any kind of prefix/root. They just get extracted as is.
> I wonder what pip does in that case. I'd guess it just gets ignored.

I see. I think pip uses them to find the directory to which they are extracted.

**Comment by lazka** [comment]
Yeah, true.

File: mingw-w64-python-install/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+build() {
+  msg "Python build for ${CARCH}"  
+  cd "${srcdir}/python-build-${CARCH}"
+  ${MINGW_PREFIX}/bin/python setup.py build
+}
+
+package() {
+  cd "${srcdir}/python-build-${CARCH}"
+  MSYS2_ARG_CONV_EXCL="--prefix=;--install-scripts=;--install-platlib=" \
```
Comment (lazka) [inline mingw-w64-python-install/PKGBUILD:37]:
```suggestion
  cd "${srcdir}/python-build-${CARCH}"
  PYTHONOPTIMIZE=1 \
  MSYS2_ARG_CONV_EXCL="--prefix=;--install-scripts=;--install-platlib=" \
```

File: mingw-w64-python-typing_extensions/PKGBUILD [hunk 2]
```diff
@@ -26,14 +30,19 @@ prepare() {
 build() {
   msg "Python build for ${CARCH}"
   cd "${srcdir}/python-build-${CARCH}"
-  ${MINGW_PREFIX}/bin/python setup.py build
+  ${MINGW_PREFIX}/bin/python -m build --wheel --skip-dependency-check --no-isolation
 }
 
 package() {
   cd "${srcdir}/python-build-${CARCH}"
   MSYS2_ARG_CONV_EXCL="--prefix=;--install-scripts=;--install-platlib=" \
```
Comment (lazka) [inline mingw-w64-python-typing_extensions/PKGBUILD:38]:
```suggestion
  cd "${srcdir}/python-build-${CARCH}"
  PYTHONOPTIMIZE=1 \
  MSYS2_ARG_CONV_EXCL="--prefix=;--install-scripts=;--install-platlib=" \
```

Comment (naveen521kk) [reply mingw-w64-python-typing_extensions/PKGBUILD:38]: What does this do? Just curious.

Comment (lazka) [reply mingw-w64-python-typing_extensions/PKGBUILD]: to generate the same set of pyc files that we had previously

Comment (lazka) [reply mingw-w64-python-typing_extensions/PKGBUILD]: oh, that now generates .opt-1.pyc, but no .pyc anymore, which is worse. sorry :/

Comment (lazka) [reply mingw-w64-python-typing_extensions/PKGBUILD:38]: let's go with the initial case then.. it's unlikely that users will notice.

**Pull request merged**